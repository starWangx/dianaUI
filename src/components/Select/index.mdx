---
name: Select —— 选择框
route: /select
---
import { Playground } from 'docz'
import Select from './index.tsx'

# Select 选择框组件

## Demo
<Playground>
  <Select
    multiple
    theme='basic'
    renderOption={[
      'option1',
      'option2',
      'option3',
      'option4',
      'option5'
    ]}
  />

  <Select
    theme='calendar'
  />

  <br />
  <br />
  <Select
    theme='search'
    renderOption={[
          '猴子',
          '猪猪',
          '狗狗',
          '猫猫',
          '哈哈'
        ]}
  />

  <br />
  <br />
  <Select
    multiple
    placeholder='多选+搜索'
    theme='search'
    renderOption={[
          'option1',
          'option2',
          'option3',
          'option4',
          'option5'
        ]}
  />

  <br />
  <br />
  <Select
    multiple
    placeholder='单选+搜索'
    theme='search'
    renderOption={[
          'option1',
          'option2',
          'option3',
          'option4',
          'option5'
        ]}
  />

  <br />
  <br />
  <Select
    disabled
    theme='search'
    renderOption={[
          'option1',
          'option2',
          'option3',
          'option4',
          'option5'
        ]}
  />
</Playground>

## API - Props
- `theme` 三种主题：`basic`(默认)、`calendar`(日历)、`search`(搜索)
- `renderOption` 选择框选项渲染
- `placeholder` 占位符
- `searchPlaceholder` 搜索下拉菜单占位符
- `onSearch` 文本框值变化时回调
- `onFocus` 搜索框获得焦点时回调
- `onBlur` 搜索框失去焦点时回调
- `onChange` 选择改变的回调
- `onDelete` 删除多选标签时回调
- `onSelect` 选择选项时的回调

---

## Option 上传组件选项
## API - Props
- `title`
- `label`
- `value`
- `key`
- `className`
- `disabled`
- `opt`
- `prefix`
- `index`
- `optActive`
- `props`
- `selected`
- `onClick`
- `onMouseOver`

