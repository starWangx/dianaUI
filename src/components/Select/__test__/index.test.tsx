import * as React from 'react';
import { configure, shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Select from '../index';

configure({ adapter: new Adapter() });

describe('Select', () => {
  it('renders correctly - shallow', () => {
    const wrapper = shallow(<Select />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly - render', () => {
    const wrapper = render(<Select />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly - mount', () => {
    const wrapper = mount(<Select />);

    expect(wrapper).toMatchSnapshot();
  });
});
