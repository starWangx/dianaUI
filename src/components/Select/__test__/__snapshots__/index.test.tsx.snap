// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders correctly - mount 1`] = `ReactWrapper {}`;

exports[`Select renders correctly - render 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "style": "display:inline-block",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "sc-EHOje iQUFER hupu-select-container",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "hupu-select-wrapper",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "hupu-select-placeholder ",
                },
                "children": Array [
                  Object {
                    "data": "请选择",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Select renders correctly - shallow 1`] = `ShallowWrapper {}`;
